src/metabase/api/database.clj

```diff
@@ -64,6 +64,11 @@
      #(u/ignore-exceptions (driver/the-driver %))]]
    (deferred-tru "value must be a valid database engine.")))
 
+(defn- add-conn-uri-sensitive-exception
+  "Add redaction exception for `:conn-uri` detail, which is handled in `mi/to-json :model/Database` implementation."
+  [db]
+  (update db :details vary-meta assoc :sensitive-exceptions #{:conn-uri}))
```

Comment:
Is this stuff really even needed, given that `:conn-uri` is already in `default-sensitive-fields`? I'm a little itchy about using metadata to pass around which fields are sensitive since it's really easy to forget not to do this in other endpoints and it's not super obvious how this code works since it sort of splits up the responsible code for redacting stuff between the API namespace and the model namespace. I rather just always do it in the model code so it lives in one place and you can't forget to add it